backend/src/GraphQl/OpenIdConnect/Application/ApplicationMutations.cs

```diff
@@ -0,0 +1,259 @@
+                    new[] { nameof(input), nameof(input.ClientId).FirstCharToLower() }
+                )
+            );
+        }
+        var descriptor = new OpenIddictApplicationDescriptor
+        {
+            ClientId = input.ClientId,
+            ClientSecret = "application.ClientSecret",
+            DisplayName = input.DisplayName,
+            ConsentType = environment.IsEnvironment(Program.TestEnvironment)
+                        ? OpenIddictConstants.ConsentTypes.Systematic
+                        : OpenIddictConstants.ConsentTypes.Explicit,
+            RedirectUris =
+            {
+                new Uri(environment.IsEnvironment(Program.TestEnvironment)
+                    ? "urn:test"
+                    : input.RedirectUri,
+                    UriKind.Absolute)
+            },
+            PostLogoutRedirectUris =
+            {
+                new Uri(environment.IsEnvironment(Program.TestEnvironment)
+                    ? "urn:test"
+                    : input.PostLogoutRedirectUri,
+                    UriKind.Absolute)
+            },
+            Permissions =
+            {
+                // Add default permissions
+                OpenIddictConstants.Permissions.Endpoints.Authorization,
+                OpenIddictConstants.Permissions.Endpoints.Introspection,
+                OpenIddictConstants.Permissions.Endpoints.Logout,
+                OpenIddictConstants.Permissions.Endpoints.Revocation,
+                OpenIddictConstants.Permissions.Endpoints.Token,
+                environment.IsEnvironment(Program.TestEnvironment)
+                    ? OpenIddictConstants.Permissions.GrantTypes.Password
+                    : OpenIddictConstants.Permissions.GrantTypes.AuthorizationCode,
+                OpenIddictConstants.Permissions.GrantTypes.RefreshToken,
+                environment.IsEnvironment(Program.TestEnvironment)
+                    ? OpenIddictConstants.Permissions.ResponseTypes.Token
+                    : OpenIddictConstants.Permissions.ResponseTypes.Code,
+                OpenIddictConstants.Permissions.Scopes.Address,
+                OpenIddictConstants.Permissions.Scopes.Email,
+                OpenIddictConstants.Permissions.Scopes.Phone,
+                OpenIddictConstants.Permissions.Scopes.Profile,
+                OpenIddictConstants.Permissions.Scopes.Roles,
+            },
+            Requirements =
+            {
+                OpenIddictConstants.Requirements.Features.ProofKeyForCodeExchange
+            }
+        };
+
+        foreach (var permission in JsonSerializer.Deserialize<List<string>>(input.Permissions) ?? new List<string>())
+        {
+            descriptor.Permissions.Add(permission);
+        }
+
+        var application = await applicationManager.CreateAsync(descriptor, cancellationToken).ConfigureAwait(false);
+        var institution = await institutionById.LoadAsync(input.AssociatedInstitutionId, cancellationToken).ConfigureAwait(false);
```

Comment:
If the associated institution is `null`, then this mutation should fail and no application be generated in the first place. Each application should be associated with at least one institution.